pkg/storaged/drive/drive.jsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+import client from "../client";
+
+import { CardBody } from "@patternfly/react-core/dist/esm/components/Card/index.js";
+import { DescriptionList } from "@patternfly/react-core/dist/esm/components/DescriptionList/index.js";
+import { Flex } from "@patternfly/react-core/dist/esm/layouts/Flex/index.js";
+import { OutlinedHddIcon, MicrochipIcon, SaveIcon } from "@patternfly/react-icons";
+
+import { StorageCard, StorageDescription, new_card, new_page } from "../pages.jsx";
+import { block_name, drive_name, format_temperature, fmt_size_long } from "../utils.js";
+import { make_block_page } from "../block/create-pages.jsx";
+import { partitionable_block_actions } from "../partitions/actions.jsx";
+
+const _ = cockpit.gettext;
+
+export function make_drive_page(parent, drive) {
+    let block = client.drives_block[drive.path];
+
+    if (!block) {
+        // A drive without a primary block device might be
+        // a unconfigured multipath device.  Try to hobble
+        // along here by arbitrarily picking one of the
+        // multipath devices.
+        block = client.drives_multipath_blocks[drive.path][0];
+    }
+
+    if (!block)
+        return;
+
+    let cls;
+    if (client.drives_iscsi_session[drive.path])
+        cls = "iscsi";
+    else if (drive.MediaRemovable || drive.Media)
+        cls = "media";
+    else
+        cls = (drive.RotationRate === 0) ? "ssd" : "hdd";
+
+    const drive_title = {
+        media: _("Media drive"),
+        ssd: _("Solid State Drive"),
+        hdd: _("Hard Disk Drive"),
+        iscsi: _("iSCSI Drive"),
+    };
+
+    const drive_icon = {
+        media: SaveIcon,
+        ssd: MicrochipIcon,
+        hdd: OutlinedHddIcon,
```

Comment:
The toplevel iSCSI session/portal/target has a network icon, but the individual drives have not. I don't think it would add much to give them all the same icon, and would rather add clutter.

![image](https://github.com/cockpit-project/cockpit/assets/3228183/a22b09e2-04f1-4874-b1a7-fa7326a9c0df)